### ISSUE TITLE
Different “extra” fields for different appointment pages

### ISSUE THREAD

**Issue by SteMautic** [SergeyMosin/Appointments#477]
Hi there,
I love the appointments' app, especially with the ability to add extra fields, as mentioned in [https://github.com/SergeyMosin/Appointments/issues/223](url).

Is there a way to have different extra fields per appointment booking pages?
It would be absolutely fine if that is possible to be set via the JSON entry. I have different appointment forms for different contacts, which need different questions to be answered when booking an appointment.

Would be a nice to have.
Thanks,
Stephan

**Comment by SergeyMosin** [comment]
It will be possible in v2 , which is scheduled to for beta release soon.

**Comment by SteMautic** [comment]
Sounds great, Sergey!

**Comment by SteMautic** [comment]
Hi Sergey,
I just updated to version 2.0.2 of the Appointments app and wonder why it only saves one extra field. 
Are there any limits to the number of fields? Could I use several fields with the same type?
For one of my appointment booking pages, I need to ask about 13 questions. But even if I only add 3 extra fields (one of each type), only the first one is saved to the event description and send to my email address.
I checked the event description field could hold more than 8000 characters, which should be more than enough.

There are some entries in my log file, which read like the following:
..."user":"--","app":"PHP","method":"GET","url":"/apps/appointments/pub/UbnqVzkLE3Xe3dQ%3D/form","message":"Return type of OCA\\Appointments\\Backend\\FakeIterator::current() should either be compatible with Iterator::current(): mixed, or the #[\\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice at /pathtonextcloud/apps/appointments/lib/Backend/FakeIterator.php#54","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0","version":"28.0.3.2","data":{"app":"PHP"}}

There are four entries with FakeIterator::current(), FakeIterator::next, (), FakeIterator::key() , FakeIterator::valid(), and FakeIterator::rewind(). After these entries I see the following:

..."app":"dav","method":"PUT","url":"/ocs/v2.php/apps/user_status/api/v1/heartbeat?format=json","message":"Found 1 applicable event(s), changing user status","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36","version":"28.0.3.2","data":{"app":"dav","user":"username"}}
